core/src/main/scala/com/typesafe/tools/mima/lib/analyze/method/MethodChecker.scala

```diff
@@ -34,15 +35,21 @@ private[analyze] abstract class BaseMethodChecker extends Checker[MethodInfo, Cl
     }
   }
 
-  private def methSigCheck(oldmeth: MethodInfo, newMeth: MethodInfo): Boolean = {
-    oldmeth.signature == newMeth.signature || (
-      newMeth.bytecodeName == MemberInfo.ConstructorName && newMeth.signature.isEmpty
-    )
-  }
-
   private def uniques(methods: List[MethodInfo]): List[MethodInfo] =
     methods.groupBy(_.parametersDesc).values.map(_.head).toList
 }
+private[analyze] object BaseMethodChecker {
+  def hasMatchingDescriptorAndSignature(oldMethod: MethodInfo, newMethod: MethodInfo): Boolean =
+    oldMethod.descriptor == newMethod.descriptor && hasMatchingSignature(oldMethod.signature, newMethod.signature, newMethod.bytecodeName)
+
+  // Assumes it is already checked that the descriptors match
+  def hasMatchingSignature(oldSignature: String, newSignature: String, bytecodeName: String): Boolean =
+  oldSignature == newSignature ||
+    // Special case for https://github.com/scala/scala/pull/7975:
+    (bytecodeName == MemberInfo.ConstructorName &&
+      (newSignature.isEmpty || (oldSignature.endsWith(newSignature.drop(2))))
```

Comment:
What 2 chars are dropped?

Reply:
It's now 1 char - the leading `(` of the new signature (which may have dropped the first parameter compared to the old).